pandas/core/indexes/base.py

```diff
@@ -5736,6 +5747,11 @@ def map(self, mapper, na_action=None):
             # empty
             attributes["dtype"] = self.dtype
 
+        if self._is_backward_compat_public_numeric_index and is_numeric_dtype(
```

Comment:
why can't this be handled in the Index constructor itself?

Reply:
It's the same thing as the others: The Index constructor returns Int64Index etc. in order to maintain backward compat.

When we remove Int64Index etc. this block can be removed, see comment on line 363 regarding this.